src/ast/passes/codegen_llvm.cpp

```diff
@@ -2506,11 +2515,14 @@ void CodegenLLVM::generateProbe(Probe &probe,
   // by args builtin.
   if (probetype(current_attach_point_->provider) == ProbeType::tracepoint)
     tracepoint_struct_ = TracepointFormatParser::get_struct_name(full_func_id);
-  Function *func = Function::Create(
-      func_type, Function::ExternalLinkage, section_name, module_.get());
+
   int index = current_attach_point_->index() ?: probe.index();
-  func->setSection(
-      get_section_name_for_probe(section_name, index, usdt_location_index));
+  auto func_name = get_function_name_for_probe(name,
+                                               index,
+                                               usdt_location_index);
+  Function *func = Function::Create(
+      func_type, Function::ExternalLinkage, func_name, module_.get());
+  func->setSection("s_" + func_name);
```

Comment:
nit: should we have a wrapper around this so this prefix addition is centralized?

Reply:
That's a good idea, we may want to move to libbpf canonical section names one day and this will make the change clearer. Done.